merkletree_test.go

```diff
@@ -4,10 +4,80 @@ import (
 	"bytes"
 	"reflect"
 	"testing"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+	"golang.org/x/crypto/sha3"
 )
 
 var treeNonce = []byte("TREE NONCE")
 
+func TestOneEntry(t *testing.T) {
+	m := InitMerkleTree(&DefaultPolicies{}, treeNonce)
+
+	var commit [32]byte
+	var expect [32]byte
+
+	key := "key"
+	val := []byte("value")
+
+	if err := m.Set(key, val); err != nil {
+		t.Fatal(err)
+	}
+	m.RecomputeHash()
+
+	index := computePrivateIndex(key)
+
+	// Check empty node hash
+	h := sha3.NewShake128()
+	h.Write([]byte{EmptyBranchIdentifier})
+	h.Write(treeNonce)
+	h.Write(util.ToBytes([]bool{true}))
+	h.Write(util.IntToBytes(1))
+	h.Read(expect[:])
+	if !bytes.Equal(m.root.rightHash, expect[:]) {
+		t.Error("Wrong righ hash!",
+			"expected", expect,
+			"get", m.root.rightHash)
+	}
+
+	r, _ := m.Get(key)
+	if r == nil {
+		t.Error("Cannot find value of key:", key)
+		return
+	}
+	v := r.Value()
+	if !bytes.Equal(v, val) {
+		t.Errorf("Value mismatch %v / %v", v, val)
+	}
+
+	// Check leaf node hash
+	h = sha3.NewShake128()
```

Comment:
This is just test code: but you could do `h.Reset()` instead of creating a new instance.